packages/contracts-ops/contracts/CallBatch.sol

```diff
@@ -57,55 +97,111 @@ abstract contract CallBatch is Script {
         buffer.writeObjectClose(indent, terminal);
     }
 
-    function writeCallList(JsonWriter.Buffer memory buffer) private {
-        string memory indent = "";
-        string memory inner = indent.nextIndent();
-        string memory innerer = inner.nextIndent();
-        buffer.writeLine(indent, "{");
-        buffer.writeArrayOpen(inner, domain);
-        for (uint32 i = 0; i < calls.length; i++) {
-            writeCall(buffer, innerer, calls[i], i == calls.length - 1);
+    function writeCallList(JsonWriter.Buffer memory buffer, bool local)
```

Comment:
this should be refactored - the code in `if(local)` is exactly copied with the code in the loop over `remoteDomains`. also passing a boolean as a parameter to switch on and have different function behavior is not readable

i think you can do something like (not writing in an IDE / with compiler so this might not be exactly right)
```
function writeCallList(JsonWriter.Buffer memory buffer, GovernanceMessage.Call[] storage calls) private {
        string memory indent = "";
        string memory inner = indent.nextIndent();
        string memory innerer = inner.nextIndent();
        buffer.writeLine(indent, "{");
        buffer.writeArrayOpen(inner, domain);
        for (uint32 i = 0; i < calls.length; i++) {
            writeCall(buffer, innerer, calls[i], i == calls.length - 1);
       }
        buffer.writeArrayClose(inner, true);
        buffer.writeLine(indent, "}");
}

function writeLocal(JsonWriter.Buffer memory buffer) {
       writeCallList(buffer, localCalls);
}

function writeRemotes(JsonWriter.Buffer memory buffer) {
       for (uint256 j; j < remoteDomains.length; j++) {
              writeCallList(buffer, remoteCalls[j]);
       }
}

```